File: plugin/src/test/sc/plugin2020/util/TestJUnitUtil.java
package sc.plugin2020.util;

import static junit.framework.TestCase.assertTrue;
import static junit.framework.TestCase.fail;

/**
 * @author rra
 * @since Jul 4, 2009
 */
public class TestJUnitUtil {

  public interface Throwable {
    public void execute() throws Exception;
  }

  public static void assertThrows(Class<? extends Exception> clazz, Throwable run) {
    try {
      run.execute();
      fail();
    } catch (Exception e) {
      System.out.println(e.getMessage());
      assertTrue(clazz.isInstance(e));
    }
  }

  public static void assertThrows(Class<? extends Exception> clazz, Throwable run, String message) {
    try {
      run.execute();
      fail();
    } catch (Exception e) {
      assertTrue(clazz.isInstance(e) && e.getMessage().equals(message));
    }
  }

}

File: plugin/src/test/sc/plugin2020/util/TestJUnitUtil.kt
package sc.plugin2020.util

import junit.framework.TestCase.assertTrue
import junit.framework.TestCase.fail
import org.junit.Assert.assertEquals

object TestJUnitUtil {
    
    @JvmStatic
    fun assertContentEquals(expected: Collection<*>, actual: Collection<*>) {
        assertEquals("Size of collections does not match", expected.size, actual.size)
        assertTrue("Unexpected Elements!\n" +
                "Expected: $expected\n" +
                "Actual: $actual", expected.containsAll(actual))
        assertTrue("Missing Elements!\n" +
                "Expected: $expected\n" +
                "Actual: $actual", actual.containsAll(expected))
    }
    
    @JvmStatic
    fun assertThrows(clazz: Class<out Exception>, run: () -> Unit) {
        try {
            run()
            fail()
        } catch(e: Exception) {
            assertTrue(e.toString(), clazz.isInstance(e))
        }
        
    }
    
    @JvmStatic
    fun assertThrows(clazz: Class<out Exception>, run: () -> Unit, message: String) {
        try {
            run()
            fail()
        } catch(e: Exception) {
            assertTrue(e.toString(), clazz.isInstance(e) && e.message == message)
        }
        
    }
    
}
